src/components/MenuItem/styled/li.js

```diff
@@ -1,8 +1,17 @@
 import styled from 'styled-components';
 import { FONT_SIZE_TEXT_MEDIUM } from '../../../styles/fontSizes';
-import { COLOR_GRAY_2, COLOR_DARK_1, COLOR_GRAY_1 } from '../../../styles/colors';
+import getTheme from '../../../styles/helpers/getTheme';
 
-const StyledLi = styled.li`
+const StyledLi = styled.li.attrs(props => {
+    const palette = getTheme(props).palette;
+    const { main: mainText, disabled: disabledText } = palette.text;
+    const hover = palette.action.hover;
+    return {
+        mainText,
+        hover,
+        disabledText,
+    };
+})`
```

Comment:
since we are going to be repeating this in every styled component, can we create kinda of hoc that dynamically does the work??

Reply:
```
const SyliedLi = attachThemeAttr(styled.li)`
    color: ${props => props.palette.text.main};
`;
```

```
function attachThemeAttrs(styledElement) {
    return atyledElement.attrs((props) => {
        const theme = getTheme(props);
        return {
             ...props,
             ...theme
        }
    })
}
```